hydra-tui/src/Hydra/TUI/Forms.hs

```diff
@@ -9,40 +9,42 @@ import Hydra.Prelude hiding (Down, State, padLeft)
 
 import Hydra.Cardano.Api
 
+import Brick (BrickEvent (..), vBox, withDefAttr)
 import Brick.Forms (
-  Form,
-  checkboxField,
+  Form (..),
+  FormField (..),
+  FormFieldState (..),
+  FormFieldVisibilityMode (..),
+  focusedFormInputAttr,
   newForm,
   radioField,
  )
+import Brick.Types (Location (..))
+import Brick.Widgets.Core (clickable, putCursor, txt, (<+>))
 import Cardano.Api.UTxO qualified as UTxO
 import Data.Map.Strict qualified as Map
+import Data.Text qualified as Text
+import Graphics.Vty (Event (..), Key (..))
 import Hydra.Chain.Direct.State ()
-import Lens.Micro (Lens', lens)
+import Lens.Micro (Lens')
 import Prelude qualified
 
 utxoCheckboxField ::
   forall s e n.
-  ( s ~ Map.Map TxIn (TxOut CtxUTxO, Bool)
+  ( s ~ Map TxIn (TxOut CtxUTxO, Bool)
   , n ~ Text
   ) =>
   Map TxIn (TxOut CtxUTxO) ->
   Form s e n
 utxoCheckboxField u =
-  newForm
-    [ checkboxField
-      (checkboxLens k)
-      ("checkboxField@" <> show k)
-      (UTxO.render (k, v))
-    | (k, v) <- Map.toList u
-    ]
-    ((,False) <$> u)
- where
-  checkboxLens :: Ord k => k -> Lens' (Map k (v, Bool)) Bool
-  checkboxLens i =
-    lens
-      (maybe False snd . Map.lookup i)
-      (\s b -> Map.adjust (second (const b)) i s)
+  let items = Map.map (,False) u
+   in newForm
+        [ checkboxGroupField '[' 'X' ']' id $
```

Comment:
Must: not use `id` as lens

Not sure, but in the past we have stumbled over using `id` as a lens for updating state as, obviously, it will not amend any value. How did you test this? Likely you want to leave the `checkboxLens`.